### PR TITLE
refactor: improve test coverage for CLI

### PR DESCRIPTION
**Pull request by IyadhKhalfallah**
---

- [ ] This PR references an open issue
- [ ] Added a test
- [ ] Updated the documentation
- [ ] Added a screenshot of a successful customer deployment

### CODE REVIEW

**Review by joschkabraun — CHANGES_REQUESTED** [review]
In `cli/__init__` test coverage is only ~50%. We can improve this by:
- adding more tests
- removing `logs` from parser (as it's removed either way
- removing `survey` from parser (as this is handled by JAC)

File: now/cli/parser.py [hunk 1]
```diff
@@ -37,25 +37,6 @@ def set_base_parser():
     return parser
 
 
-def set_help_parser(parser=None):
```
Comment (makram93) [inline now/cli/parser.py:40]: Why are we removing this?

File: now/cli/__init__.py [hunk 2]
```diff
@@ -18,13 +17,15 @@
...
+def get_run_args(parser=None):
     from now.cli.parser import get_main_parser
 
-    parser = get_main_parser()
+    if not parser:
+        parser = get_main_parser()
     if len(sys.argv) == 1:
         parser.print_help()
         exit()
+    print(parser.parse_known_args())
```
Comment (makram93) [inline now/cli/__init__.py:28]: remove this

File: now/cli/parser.py (hunk 1 above)
Comment (IyadhKhalfallah) [reply now/cli/parser.py:40]: @makram93 it is not used anywhere

File: tests/unit/test_cli.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+from now.cli import cli, get_run_args
+
+
+class TestClass:
+    def __init__(self):
+        self.example = 'test'
+
+
+class Args:
```
Comment (joschkabraun) [inline tests/unit/test_cli.py:13]: let's use `Namespace`

**Review by joschkabraun — COMMENTED** [review]
As discussed in the pairing, nice increase in code cov!
Let's make sure that the other two tasks are also tested

File: now/cli/__init__.py [hunk 4]
```diff
@@ -18,10 +17,11 @@
 # fix
 
 
-def _get_run_args():
+def get_run_args(parser=None):
```
[the remark this answers is not in the document]
Comment (IyadhKhalfallah) [reply now/cli/__init__.py:20]: You're right, I was passing the argument in a previous test then I removed it. Thanks for the catch

File: tests/unit/test_cli.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
+from argparse import ArgumentParser, Namespace
+
+from pytest_mock import MockerFixture
+
+from now.cli import cli, get_run_args
+
+
+class TestClass:
```
Comment (joschkabraun) [inline tests/unit/test_cli.py:8]: let's use `Namespace`

**Review by joschkabraun — CHANGES_REQUESTED** [review]
Small suggestion, otherwise LGTM

**Review by Tanguyabel — APPROVED** [review]
LGTM

**Pull request merged**